### PR TITLE
:lipstick: enhance search for proposal by title and speaker

### LINKED ISSUE #427: Possility to search by any fields
Right now, it is only possible to do a search by the title of a conference. 
We can maybe improve this feature, and add the ability to search by other fields. 

In the screenshot below, I am doing a search by the speaker thanks to a `speaker` prefix. 

What do you think ? @bpetetot 

image

### PR DESCRIPTION
**Pull request by corinnekrych**
Partially fixes https://github.com/bpetetot/conference-hall/issues/427
This is moving the filtering from firebase to the UI. 
As we've already fetched the speakers to display them in the list, it's easy to add a filtering field at that level.
The searchable fields for a proposal should be: title, speaker and abstract.
I haven't included search by abstract because I think it's not display in the list itself (ie: in the current view) so this filtering will not be visible. 

wdyt @bpetetot @EmmanuelDemey ?

### CODE REVIEW

**Comment by frinyvonnick** [comment]
wdyt => "What do you think ?" ❓

**Comment by corinnekrych** [comment]
yes!

File: src/screens/organizer/proposals/proposals.reactions.js [hunk 1]
```diff
@@ -48,7 +50,7 @@ export const loadProposals = async (action, store, { router }) => {
 
   const filters = store.ui.organizer.proposals.get()
   const proposals = await firebase.fetchEventProposals(eventId, uid, filters)
-  const props = await Promise.all(proposals.map(async (proposal) => {
+  let props = await Promise.all(proposals.map(async (proposal) => {
```
Comment (frinyvonnick) [inline src/screens/organizer/proposals/proposals.reactions.js:53]: maybe `props` could be renamed `filteredProposals` ?

File: src/screens/organizer/proposals/proposals.reactions.js [hunk 2]
```diff
@@ -63,6 +65,13 @@ export const loadProposals = async (action, store, { router }) => {
     prop.speakerName = speakerNameList.join(' & ')
     return prop
   }))
+  // Cross-fields search better handled in UI than firebase
+  const { search } = filters
+  if (search) {
+    const searchQuery = deburr(toLower(search))
+    props = props.filter(proposal => deburr(toLower(proposal.title)).includes(searchQuery)
```
Comment (frinyvonnick) [inline src/screens/organizer/proposals/proposals.reactions.js:72]: I guess we will add more searchable fields later. Could we think about a more generic way to add fields ?

Comment (bpetetot) [reply src/screens/organizer/proposals/proposals.reactions.js:72]: imo, I dont think it's necessary to be too much generic right now.

**Review by bpetetot — APPROVED** [review]
Thanks @corinnekrych 👍

**Review by bpetetot — APPROVED** [review]
Great thanks

**Pull request merged**